Google.Api.Generator/Generation/SnippetCodeGenerator.cs

```diff
@@ -155,7 +191,41 @@ private class SnippetDef
                 MethodGenerator(ctx, true)
                     .WithXmlDoc(XmlDoc.Summary($"Snippet for {TargetMethodName}"));
 
-            public NamespaceDeclarationSyntax StandaloneSnippet(SourceFileContext ctx)
+            private SnippetMetadata GenerateMetadata() =>
+                new SnippetMetadata
+                {
+                    Title = SnippetMethodName,
+                    Description = TargetMethod.DocLines.FirstOrDefault() ?? $"Snippet for {TargetMethodName}",
+                    Language = Language.CSharp,
+                    Canonical = Canonical,
+                    Origin = SnippetMetadata.Types.Origin.ApiDefinition,
+                    ClientMethod = new ClientMethod
+                    {
+                        ShortName = TargetMethodName,
+                        FullName = $"{TargetMethod.Svc.ClientAbstractTyp.FullName}.{TargetMethodName}",
+                        Async = !Sync,
+                        ResultType = Sync
+                            ? TargetMethod.SyncReturnTyp is Typ.VoidTyp ? "" : TargetMethod.SyncReturnTyp.FullName
+                            : TargetMethod.AsyncReturnTyp.FullName,
+                        Client = new ServiceClient
+                        {
+                            ShortName = TargetMethod.Svc.ClientAbstractTyp.Name,
+                            FullName = TargetMethod.Svc.ClientAbstractTyp.FullName
+                        },
+                        Method = new Method
+                        {
+                            ShortName = TargetMethod.ProtoRpcName,
+                            FullName = $"{TargetMethod.Svc.ServiceFullName}.{TargetMethod.ProtoRpcName}",
```

Comment:
Isn't this meant to include the Async suffix for async methods, i.e. it's the exact generated method name? (Ditto ShortName.)

Reply:
Ah right, I'm with you.